aten/src/ATen/native/cuda/SpectralOps.cu

```diff
@@ -190,8 +190,45 @@ static inline Tensor _run_cufft(
   CUFFT_CHECK(cufftSetWorkArea(plan, ws.data_ptr()));
 
   // run
+#ifdef __HIP_PLATFORM_HCC__
+  if (input.type().scalarType() == ScalarType::Float) {
```

Comment:
Can we file a bug with hipfft saying that their API is too complicated? Forty lines versus one...

Reply:
Already done internally. It may be some time till they get around to that, unfortunately.